pkg/agent/validations.go

```diff
@@ -2,138 +2,123 @@ package agent
 
 import (
 	"encoding/json"
-	"fmt"
-	"reflect"
-	"sort"
 
 	"github.com/openshift/assisted-service/api/common"
 	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
 const (
 	validationFailure string = "failure"
 	validationError   string = "error"
```

Comment:
What is the difference between 'failure' and 'error'? Is one of these actually terminal?

Reply:
Digging through the assisted service code its the different between incorrect input and a failing validation.

Here is the code I reviewed to come to that conclusion: https://github.com/openshift/assisted-service/blob/master/internal/host/validator.go

Reply:
As far as terminal.. it doesn't seem so.